src/transport/p2p.cu

```diff
@@ -493,17 +493,21 @@ static ncclResult_t getGpuHdpReg(int cudaDev, uint32_t** hdp) {
     return id;
   };
 
-  const auto& find_agent = [](hsa_agent_t agent, void* out) {
-    hsa_agent_t* found_agent = (hsa_agent_t*)out;
-    uint16_t id = (uint16_t)found_agent->handle;
+  union find_agent_args {
+    hsa_agent_t agent;
+    uint16_t id;
+  } args;
+
+  const auto& find_agent = [](hsa_agent_t agent, void* arg) {
+    uint16_t id = ((union find_agent_args *)arg)->id;
     hsa_device_type_t type;
     hsa_agent_get_info(agent, HSA_AGENT_INFO_DEVICE, (void*)&type);
     if(type == HSA_DEVICE_TYPE_GPU) {
       uint16_t bdf_id = 1;
       hsa_agent_get_info(agent, (hsa_agent_info_t)HSA_AMD_AGENT_INFO_BDFID, &bdf_id);
       if(bdf_id == id) {
-        *found_agent=agent;
-        return HSA_STATUS_ERROR;
+        ((union find_agent_args *)arg)->agent=agent;
```

Comment:
Use of union here is inconsistent with other C style typecasts.  Should this be (find_agent_args*)arg to match the other casts or is the project style okay with both uses?

Also feel free to ignore this since it's entirely a style thing.

Reply:
Nevermind.  Code already has a mix.